lib/matplotlib/colors.py

```diff
@@ -234,6 +227,16 @@ def to_rgba_array(c, alpha=None):
         if np.any((result < 0) | (result > 1)):
             raise ValueError("RGBA values should be within 0-1 range")
         return result
+    # Handle single values.
+    # Note that this occurs *after* handling inputs that are already arrays, as
+    # `to_rgba(c, alpha)` (below) is expensive for such inputs, due to the need
+    # to format the array in the ValueError message(!).
+    if isinstance(c, six.string_types) and c.lower() == "none":
+        return np.zeros((0, 4), float)
```

Comment:
Why move these first two lines as they seem to be a legitimate fast-track for certain cases?